src/content/posts/docker_volume_introduction.md

```diff
@@ -0,0 +1,84 @@
+
+A docker's volume is the action of **mounting an existing file or directory** of the host (the computer running the container) **into a container**. To rephrase, it enables a container to **share a memory space with the host**. So both the container and the user can access and modify it in real time and because it's not "contained" in the container but only mounted, it'll **persist after the container's death**.
+
+## Why do we need volumes?
+
+Docker containerization is very useful on lots of points for development but it's very annoying if your application **need to produce persistent data**, like logs. Indeed if you **kill** your container (to update it for example) or if it **crashes**, you **lose everything that where inside** and understanding why your application crashed without its logs can be very complicated or even impossible.
+
+So we need to find a way to **keep certain files after the container's death** and that can be **achieved with volumes**!.
+
+## How to create a volume?
+
+To create a volume, you need to add a specific argument to the `docker run` command.
+
+> You can also create volumes outside of containers with [`docker volume`](https://docs.docker.com/storage/volumes/#create-and-manage-volumes)
+
+There is two different arguments which are really similar in their use. Those are `--mount` and `--volume`, both can be used to create volumes (or bind mounts, see Disclaimer above).  
+However they slightly differ: if you enter a non existing file to mount, `--volume` will create it when `--mount` will return an error, also `--volume` needs fewer arguments. For these reason and because we focused on volumes in this post we'll describe the `--volume` argument. 
+
+The exact syntax is:
+```bash
+docker run <other args> --volume PATH_TO_HOST_FILE:PATH_TO_CONTAINER_FILE:<optional args>
+```
+
+> Note that you can use `-v` instead of `--volume`.
+
+> Note that you can mount a directory instead of a file with the same method.
+
+> A similar syntax exists for [`docker-compose.yml`](https://docs.docker.com/compose/compose-file/compose-file-v3/#volumes)
+
+Optional args must be separated by comas, they are not used very often but lets detail the most useful one quickly :
+- `ro` : if used, the mounted files/folders will be in read only mode inside the container
+
+A concrete example with the [code of our RootMe discord bot](https://github.com/iScsc/RootPythia/blob/2681ca26286ea5063371536e995a5e3cf39734a5/run.sh#L12):
+```bash
+    source ./.env.prod
+
+    docker run --rm --interactive --tty \
+	--detach \
+	--volume $(realpath -P ${LOG_FOLDER}):/opt/${NAME}/logs \
+	--env-file .env.prod \
+	--name ${NAME} \
+	${NAME}:latest
+```
+In this example `LOG_FOLDER` is an environment variable defined in the `.env.prod` file.  
+The code will written transparently to `/opt/${NAME}/logs/<log file>` but because this folder is shared with the host, logs will **be available in the host** and survive potential application crashes or updates.
```

Comment:
```suggestion
The code will write transparently to `/opt/${NAME}/logs/<log file>` but because this folder is shared with the host, logs will **be available in the host** and survive potential application crashes or updates.
```

Reply:
Or 'will be written' maybe ?

Reply:
nop 'write' is expected indeed!

Reply:
[9ff9c64](https://github.com/iScsc/blog.iscsc.fr/pull/28/commits/9ff9c6474f45d0158cfe76218d6d932b68a02cd9)